### ISSUE TITLE
Add location to notes

### ISSUE THREAD

**Issue by vHanda** [GitJournal/GitJournal#187]
Optionally, add the user's current location to the notes. Saving the "latitude" and "longitude" is simple, however I'm not sure how to do a reverse lookup without using a 3rd party service and thereby compromising privacy.

I could use [GeoNames](http://www.geonames.org/export/web-services.html#countrysubdiv) either through their API, or just download the `cities15000.zip` data set which is only 2.9 mbs. It could be downloaded on demand and the reverse lookup could be done completely offline.

---
> Please vote on this issue with a 👍 reaction. It's one of the primary mechanisms used to prioritize issues. Please also subscribe to the issue if you're interested, as that way I can discuss possible implementations and share mockups.

**Comment by bisraelsen** [comment]
I have existing notes with lat/long metadata. It would be excellent to allow for this metadata field (and possibly do something with it in the app). I'm not an app developer, but I thought you could just allow an app to access the device location if the user gives permission. When you say "reverse lookup" are you referring to finding an address from lat/long coordinates?

**Comment by vHanda** [comment]
> I have existing notes with lat/long metadata. It would be excellent to allow for this metadata field (and possibly do something with it in the app). I'm not an app developer, but I thought you could just allow an app to access the device location if the user gives permission. When you say "reverse lookup" are you referring to finding an address from lat/long coordinates?

Yup. Exactly. Mapping the latitude and longitude to a particular City.
